SPDX validation in npm

othiym23 opened the pull request:
Going to also land fixed versions of npm-controlled dependencies with invalid SPDX declarations on this branch. Needs a test for the `normalize-package-data` change to ensure that this doesn't ever cause installation to fail, which would make this a semver-major change.

_r_: @kemitchell

othiym23 committed:
normalize-package-data@2.1.0

Incorporates validation of `package.json` license fields against the
SPDX license expression syntax, raising warnings if the license is
missing or doesn't parse as valid SPDX.

othiym23 committed:
init-package-json@1.5.0

Validate the license field upon new package creation against SPDX.

othiym23 committed:
src: dedupe spdx up a level

othiym23 committed:
semver@4.3.4

Switch to ISC from BSD (and incidentally an SPDX-compatible licensing
stanza).

kemitchell commented:
You may find [license-checker](http://npmjs.com/packages/license-checker) by @davglass useful.

othiym23 commented:
No need – with the new `normalize-package-data` installed, all I need to do is run `npm i` and I get a near-overwhelming list of things I need to clean up. ;)

kemitchell commented:
I haven't `npm link`d it in yet.

Yeah, it's a mess out there, but nothing quite motivates a geek like avoidable junk on `stderr`.

othiym23 committed:
once@1.3.2

Swtich from BSD to ISC licenses, and in the process adopt an
SPDX-compliant license string.
